File: Tom/Level1/rinfo_collect.cpp
#include <string>
#include <vector>
using namespace std;

bool func(string today ,vector<string> terms, string p) {
    char t = p[p.length() - 1]; // 동의한 약관 알파벳
    string period; // 약관 기간

    for (int i = 0; i < terms.size(); i++) {
        if (t == terms[i][0]) {
            for (int j = 2; j < terms[i].length(); j++)
                period += terms[i][j];
        }
    }

    // 각 동의한 약관 날짜에서 년, 월, 일 int형으로 추출
    int year = stoi(p.substr(0, 4));
    int month = stoi(p.substr(5, 2));
    int date = stoi(p.substr(8, 2));

    // 동의 날짜 + 약관 기간
    month += stoi(period);

    // 12월까지 있으니까, 12월 넘어가는거 계산 (주의, month가 12 이하가 되도록 계속 빼줘야댐!)
    while (month > 12) {
        month -= 12;
        year++;
    }

    // 현재 날짜에서 년, 월, 일 int형으로 추출
    int t_year = stoi(today.substr(0, 4));
    int t_month = stoi(today.substr(5, 2));
    int t_date = stoi(today.substr(8, 2));

    // 현재년도 > 유효년도
    if (t_year > year) return false;
    // 같은 년인 경우, month 비교도 필요함....
    else if (t_year == year && t_month > month) return false;
    // 같은 년, 월인 경우, date 비교도 필요함....
    else if (t_year == year && t_month == month && t_date >= date) return false;

    return true;
}

vector<int> solution(string today, vector<string> terms, vector<string> privacies) {
    vector<int> answer;

    for (int i = 0; i < privacies.size(); i++)
        if (!func(today, terms, privacies[i]))
            answer.push_back(i + 1);
  
    return answer;
}

File: Paul/Level1/rinfo_collect.py
# 개인정보 수집 유효기간

def dateToDay(date):
    year, month, day = map(int, date.split("."))

    return (year * 12 * 28) + (month * 28) + day

def solution(today, terms, privacies):

    answer = []
    terms_two = []
    privacies_two = []
    count = 0

    for index in privacies:
        privacies_two.append(list(map(str, index.split(" "))))

    for index in terms:
        terms_two.append(list(map(str, index.split(" "))))

    for index_privacies in privacies_two:
        count += 1

        for index_terms in terms_two:

            if index_terms[0] == index_privacies[1]:
                temp_list = list(map(int, index_privacies[0].split(".")))
                terms_days = int(index_terms[1]) * 28
                temp_list[2] += terms_days - 1

                if temp_list[2] + temp_list[1] * 28 + temp_list[0] * 12 * 28 < dateToDay(today):
                    answer.append(count)

    return answer

print(solution("2022.05.19", ["A 6", "B 12", "C 3"], ["2021.05.02 A", "2021.07.01 B", "2022.02.19 C", "2022.02.20 C"]))
print(solution("2020.01.01", ["Z 3", "D 5"], ["2019.01.01 D", "2019.11.15 Z", "2019.08.02 D", "2019.07.01 D", "2018.12.28 Z"]))
